spec/requests/api/v2/providers/suggest_provider_spec.rb

```diff
@@ -3,74 +3,78 @@
 describe "GET /suggest" do
   let(:jsonapi_renderer) { JSONAPI::Serializable::Renderer.new }
   let(:organisation) { create(:organisation) }
-  let(:provider1) { create(:provider, provider_name: "PROVIDER 1", organisations: [organisation]) }
-  let(:provider2) { create(:provider, provider_name: "PROVIDER 2", organisations: [organisation]) }
   let(:next_recruitment_cycle) { find_or_create(:recruitment_cycle, :next) }
-  let(:provider3) { create(:provider, provider_name: provider2.provider_name, organisations: [organisation], recruitment_cycle: next_recruitment_cycle) }
-  let(:provider4) { create(:provider, organisations: [organisation], recruitment_cycle: next_recruitment_cycle) }
   let(:user) { create :user, organisations: [organisation] }
   let(:payload) { { email: user.email } }
   let(:token) { build_jwt :apiv2, payload: payload }
   let(:credentials) do
     ActionController::HttpAuthentication::Token.encode_credentials(token)
   end
 
-  before do
-    provider1
-    provider2
-    provider3
-    provider4
-  end
-
-  it "searches for a particular provider for the user in the current recruitment cycle" do
-    get "/api/v2/providers/suggest?query=#{provider2.provider_name}",
-        headers: { "HTTP_AUTHORIZATION" => credentials }
+  context "current recruitment cycle" do
+    it "searches for a particular provider" do
+      provider = create(:provider, provider_name: "PROVIDER 1", organisations: [organisation])
+      create(:provider, provider_name: "PROVIDER 2", organisations: [organisation])
```

Comment:
I totally missed, after the first few glances, that we are in fact creating two providers here. What if we gave this a name, like `_provider_missing_from_results`?